File: exercise/php/day15/src/SantaChristmasList/Dependencies/Inventory.php
<?php

namespace SantaChristmasList\Dependencies;

use SantaChristmasList\Models\Gift;

class Inventory extends \ArrayObject
{
    public function pickUpGift(string $barCode): ?Gift
    {
        return $this->offsetExists($barCode) ? $this->offsetGet($barCode) : null;
    }

    public function set(string $string, Gift $toy): void
    {
        $this->offsetSet($string, $toy);
    }
}

File: exercise/java/day15/src/main/java/santaChristmasList/operations/dependencies/Inventory.java
package santaChristmasList.operations.dependencies;

import santaChristmasList.operations.models.Gift;

import java.util.HashMap;

public class Inventory extends HashMap<String, Gift> {
    public Gift pickUpGift(String barCode) {
        return this.getOrDefault(barCode, null);
    }
}